tools/toolchains/arm.py

```diff
@@ -368,6 +369,14 @@ def __init__(self, target, *args, **kwargs):
         if target.core not in self.SUPPORTED_CORES:
             raise NotSupportedException(
                 "this compiler does not support the core %s" % target.core)
+        if CORE_ARCH[target.core] < 8:
+            self.notify.cc_info({
+                'severity': "Error", 'file': "", 'line': "", 'col': "",
+                'message': "ARMC6 does not support ARM archetecture v{}"
```

Comment:
`archetecture ` typo

Reply:
What's the reason for throwing this error when using ARM Compiler 6 with v6m or v7m target?

From the compiler's view, the message could be misleading. Becuase ARM compiler officially claims the support for architectures from Armv6m and all the later ones.

Reply:
We don't test that combination.